src/other/index.jl

```diff
@@ -422,7 +422,7 @@ Base.@propagate_inbounds SubIndex(parent::AbstractIndex, cols) =
     SubIndex(parent, parent[cols])
 
 Base.length(x::SubIndex) = length(x.cols)
-Base.names(x::SubIndex) = string.(_names(x))
+Base.names(x::SubIndex) = string.(_names(x))::Vector{String}
```

Comment:
do you know why Julia fails to infer this without an annotation?

Reply:
SubIndex has `I<:AbstractIndex, S<:AbstractVector{Int}, T<:AbstractVector{Int}` as type parameters. When it doesn't know what `I, S, T` are (which is what happens when it's a `Vector{AbstractDataFrame}`, inference knows almost nothing of value. In principle an `AbstractIndex` could do *anything*. You/we could consider adding an interface: https://timholy.github.io/SnoopCompile.jl/stable/snoopr/#Inferrable-field-access-for-abstract-types-1.